clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/RemoteCacheContainerBuilder.java

```diff
@@ -0,0 +1,96 @@
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.clustering.infinispan.subsystem.remote;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.Configuration;
+import org.jboss.as.clustering.controller.ResourceServiceBuilder;
+import org.jboss.as.clustering.infinispan.InfinispanLogger;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.Service;
+import org.jboss.msc.service.ServiceBuilder;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceName;
+import org.jboss.msc.service.ServiceTarget;
+import org.wildfly.clustering.infinispan.spi.InfinispanRequirement;
+import org.wildfly.clustering.infinispan.spi.RemoteCacheContainer;
+import org.wildfly.clustering.service.InjectedValueDependency;
+import org.wildfly.clustering.service.SuppliedValueService;
+import org.wildfly.clustering.service.ValueDependency;
+
+/**
+ * @author Radoslav Husar
+ */
+public class RemoteCacheContainerBuilder implements ResourceServiceBuilder<RemoteCacheContainer> {
+
+    private ValueDependency<Configuration> configuration;
+    private final PathAddress address;
+    private final String name;
+
+    public RemoteCacheContainerBuilder(PathAddress address) {
+        this.address = address;
+        this.name = address.getLastElement().getValue();
+    }
+
+    @Override
+    public ServiceName getServiceName() {
+        return RemoteCacheContainerResourceDefinition.Capability.CONTAINER.getServiceName(this.address);
+    }
+
+    @Override
+    public RemoteCacheContainerBuilder configure(OperationContext context, ModelNode model) throws OperationFailedException {
+        this.configuration = new InjectedValueDependency<>(InfinispanRequirement.REMOTE_CONTAINER_CONFIGURATION.getServiceName(context, this.name), Configuration.class);
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder<RemoteCacheContainer> build(ServiceTarget target) {
+        Function<RemoteCacheManager, RemoteCacheContainer> mapper = remoteCacheManager -> new ManagedRemoteCacheContainer(this.name, remoteCacheManager);
+
+        Supplier<RemoteCacheManager> supplier = () -> {
+            Configuration configuration = this.configuration.getValue();
+            RemoteCacheManager remoteCacheManager = new RemoteCacheManager(configuration);
+            remoteCacheManager.start();
+            InfinispanLogger.ROOT_LOGGER.debugf("%s remote cache container started", this.name);
+            return remoteCacheManager;
+        };
+
+        Consumer<RemoteCacheManager> destroyer = remoteCacheManager -> {
```

Comment:
You can remove this lambda by having RemoteCacheContainerBuilder implement Consumer\<RemoteCacheManager\>